ai_guidelines.md

```diff
@@ -0,0 +1,139 @@
+# Guidelines for CommCare HQ
+
+This document outlines recommended practices for working with the CommCare HQ
+codebase. These guidelines apply in the following circumstances:
+
+- Reviewing code or suggesting changes
+- Writing code and making changes
+- Assisting with version control
+
+## General Principles
+
+- Focus on clarity, maintainability, and performance
+- Consider both immediate needs and long-term implications
+
+## Communication
+
+### For the AI Assistant
+
+- The AI Assistant should maintain professional, focused communication.
+- The AI Assistant should ask clarifying questions whenever details are unclear,
+  or choices are not obvious.
+- When appropriate, the AI Assistant should make recommendations on how the
+  Developer can improve their prompt to get a more effective response.
+
+### For the Developer
+
+- The Developer should provide feedback to the AI Assistant to improve future
+  responses.
+- The Developer should update this document to assist the AI Assistant across
+  sessions, and to improve the interactions of other Developers.
+
+## Tech Stack
+
+- Backend: Python, Django
+- Frontend: JavaScript, HTMX, Alpine.js, Knockout.js (legacy)
+- Databases: PostgreSQL, Elasticsearch, CouchDB (legacy)
+- Asynchronous task queue: Celery
+- Cache & message broker: Redis
+- Stream processing: Kafka
+- Version Control: Git
+
+## Writing and Reviewing Code
+
+### Code Clarity
+
+- Is the code readable and well-structured?
+- Does it follow codebase conventions?
+- Are there any potential bugs or edge cases?
+- Keep functions and methods focused on a single responsibility. Break up
+  longer functions where appropriate.
+
+### Performance
+
+- Are there any potential performance bottlenecks?
+- For database operations, are queries optimized?
+- For front-end code, are there rendering or loading concerns?
+
+### Maintainability
+
+- Have comments or docstrings that are affected by the change been updated?
+- Document complex logic, algorithms, and business rules.
+- Not all functions, classes and methods ought to have docstrings. Where
+  functionality is not obvious just from the name, a docstring should be
+  present and helpful. For usage examples, include one, or at most two, short
+  doctests where appropriate.
+- Comments should explain "why" rather than "what": As much as possible,
+  "what" should be clear from the code.
+- Keep module README files and project documentation under `docs/`
+  up-to-date.
+
+### Security
+
+- Are there any security vulnerabilities?
+- Is user input properly validated and sanitized?
+- Are permissions and access controls properly implemented?
```

Comment:
I'm not sure about having questions in this. It makes sense for code review but less so for code generation. I expect code review would be fine with affirmative statements too e.g. Be mindful of security and follow best practices for handling user input.